dataset/src/main/kotlin/com/cosmotech/dataset/azure/DatasetServiceImpl.kt

```diff
@@ -159,6 +161,32 @@ internal class DatasetServiceImpl(
     TODO("Not yet implemented")
   }
 
+  override fun searchDatasets(organizationId: String, datasetSearch: DatasetSearch): List<Dataset> {
+    var searchList: Set<Dataset> = setOf()
+    if (datasetSearch.datasetTags == null) {
+      throw CsmClientException("Dataset tag list is mandatory for search")
+    }
+
+    datasetSearch.datasetTags?.forEach { tag ->
```

Comment:
As seen together, it would be more efficient, performance-wise, to issue a single SQL Query using logical `AND` operators, instead of `N` queries.

Reply:
Also, I think transforming `datasetTags` into a `Set` would prevent useless operations against the remote data store.

Reply:
Also, I think transforming `datasetTags` into a `Set` would prevent useless operations against the remote data store.

Reply:
done